src/elements/content-sharing/hooks/useSharedLink.js

```diff
@@ -0,0 +1,138 @@
+// @flow
+import * as React from 'react';
+import noop from 'lodash/noop';
+import API from '../../../api';
+import { ACCESS_COLLAB, ACCESS_NONE, TYPE_FILE, TYPE_FOLDER } from '../../../constants';
+import { CONTENT_SHARING_SHARED_LINK_UPDATE_PARAMS } from '../constants';
+import type { RequestOptions } from '../../../common/types/api';
+import type { BoxItemPermission, ItemType } from '../../../common/types/core';
+import type { ContentSharingHooksOptions, SharedLinkUpdateLevelFnType, SharedLinkUpdateSettingsFnType } from '../types';
+
+/**
+ * Generate CRUD functions for shared links.
+ *
+ * @param {API} api
+ * @param {string} itemID
+ * @param {ItemType} itemType
+ * @param {BoxItemPermission} permissions
+ * @param {string} accessLevel
+ * @param {ContentSharingHooksOptions} [options]
+ */
+function useSharedLink(
+    api: API,
+    itemID: string,
+    itemType: ItemType,
+    permissions: ?BoxItemPermission,
+    accessLevel: string,
+    options: ContentSharingHooksOptions = {},
+) {
+    const [onAddLink, setOnAddLink] = React.useState<null | SharedLinkUpdateLevelFnType>(null);
+    const [onRemoveLink, setOnRemoveLink] = React.useState<null | SharedLinkUpdateLevelFnType>(null);
+    const [
+        changeSharedLinkAccessLevel,
+        setChangeSharedLinkAccessLevel,
+    ] = React.useState<null | SharedLinkUpdateLevelFnType>(null);
+    const [
+        changeSharedLinkPermissionLevel,
+        setChangeSharedLinkPermissionLevel,
+    ] = React.useState<null | SharedLinkUpdateLevelFnType>(null);
+    const [onSubmitSettings, setOnSubmitSettings] = React.useState<null | SharedLinkUpdateSettingsFnType>(null);
+    const [generatedFunctions, setGeneratedFunctions] = React.useState<boolean>(false);
+
+    // Storing the access level in a ref allows us to update settings, which depend on the access level, after potentially updating the access level
+    const currentAccessLevel = React.useRef(accessLevel);
+
+    const {
+        handleError = noop,
+        handleRemoveSharedLinkSuccess = arg => arg,
+        handleUpdateSharedLinkSuccess = arg => arg,
+        transformAccess = arg => arg,
+        transformPermissions = arg => arg,
+        transformSettings = (data, access) => data, // eslint-disable-line no-unused-vars
```

Comment:
is `access` included here for documentation purposes, seeing as its unused?

Reply:
Yup, otherwise Flow will complain about the additional parameter on line 109